sdk/core/azure-core/src/main/java/com/azure/core/util/Header.java

```diff
@@ -50,8 +52,13 @@ public Header(String name, String value) {
     public Header(String name, String... values) {
         Objects.requireNonNull(name, "'name' cannot be null.");
         this.name = name;
-        this.values = new LinkedList<>();
-        Collections.addAll(this.values, values);
+        int length = values.length;
+        if (length == 1) {
+            this.value = values[0];
+        } else {
+            this.values = new ArrayList<>(Math.max(length + 2, 4));
+            Collections.addAll(this.values, values);
```

Comment:
In this scenario, I wonder if we can avoid defensively copying in some cases. We might need a different ctor to indicate it, but it might be possible in many cases (e.g. when Netty is giving us an already defensively copied list) to not copy it again needlessly.

Reply:
Lets file an issue for this as this number of changes in this PR keeps growing and it'll be good to see where the new hot spots are after this change.